endpoints/openrtb2/amp_auction.go

```diff
@@ -211,6 +211,15 @@ func (deps *endpointDeps) loadRequestJSONForAmp(httpRequest *http.Request) (req
 		errs = []error{fmt.Errorf("data for tag_id '%s' includes %d imp elements. Only one is allowed", ampId, len(req.Imp))}
 		return
 	}
+
+	// Force HTTPS as AMP requires it, but pubs can forget to set it.
+	if req.Imp[0].Secure == nil {
+		secure := int8(1)
+		req.Imp[0].Secure = &secure
+	} else {
+		*req.Imp[0].Secure = 1
```

Comment:
lol I like it.

Reply:
there is the oddity that the openrtb spec sets this as a pointer, which takes up more memory than an 8 bit int, which doesn't even count the memory the int takes when the pointer is not nil.